### ISSUE TITLE
Crash when adding entity reference to entity, related to disposing of previous world after creation of new world.

### ISSUE THREAD

**Issue by pkra8** [genaray/Arch#123]
Hidden required order of world create/dispose leads to crash if not following it.
failing test flow: create new world, then dispose of old world, and it will crash 100% time when creating entity reference and adding it as component to entity. 

```
var entity1Ref = world.Reference(entity1);
entity1.Add(entity1Ref);
```
Crash/Error is:
	System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection. 	(Parameter 'index')
   at System.Collections.Generic.List`1.get_Item(Int32 index)

```
PASSING:
        for (int i = 0; i < 1000; i++) {
            var world = World.Create();

            var entity1 = world.Create(new MockComponent());
            var entity1Ref = world.Reference(entity1);
            entity1.Add(entity1Ref);

            world.Query(MockQuery, (in Entity queriedEntity, ref MockComponent mockComponent) => {
                world.Destroy(queriedEntity);
            });
            world.Dispose();
        }
```

```
FAILING:
        World oldworld = null;

        for (int i = 0; i < 1000; i++) {
            var world = World.Create();
            if (oldworld != null) {
                oldworld.Dispose();
            }
            oldworld = world;

            var entity1 = world.Create(new MockComponent());
            var entity1Ref = world.Reference(entity1);
            entity1.Add(entity1Ref);

            world.Query(MockQuery, (in Entity queriedEntity, ref MockComponent mockComponent) => {
                world.Destroy(queriedEntity);
            });

           //if you add this, test will pass:  world.Dispose();
        }
```

Mock objects
```    
private class MockComponent {
        public string Name = "Mock";
}

private QueryDescription MockQuery = new QueryDescription().WithAll<MockComponent>();
```

**Comment by genaray** [comment]
Thanks i'll look at this once my thesis is done! :)

**Comment by genaray** [comment]
Sooo i can finally look at this to fix it :) 
Have you pulled and used the master? Or did you use a nugget? If so which version?

**Comment by pkra8** [comment]
Arch 1.2.5 via nuget.

**Comment by genaray** [comment]
> Arch 1.2.5 via nuget.

And via the sources? What about Arch 1.2.5 alpha/beta? :)

**Comment by pkra8** [comment]
> > Arch 1.2.5 via nuget.
> 
> And via the sources? What about Arch 1.2.5 alpha/beta? :)

Haven't used those. I hope repro case helps nail what is happening internally down :)

**Comment by genaray** [comment]
So was this ever solved? Have you tried the latest nuget? 
Between your last version 1.2.5 and 1.2.6 there were many world related changes :)

**Comment by pkra8** [comment]
With the workaround I've made for my project I don't (and won't) have this issue.

I can run tests provided in the report to see if this was resolved when I will update to 1.2.6, but I have no promise when it happens, as I'm currently pretty occupied.
